### PR TITLE
Updating README

### LINKED ISSUE #142: [Issue] Unclear which branch should be used
### Please check all that apply

- [ ] typo
- [ ] documentation doesn't exist
- [x] documentation needs clarification
- [ ] error(s) in the example
- [ ] needs an example

### Description of the issue
The documentation does not specify which branch should be used for production environments. Furthermore, the package repository ([https://pkg.go.dev/search?q=msal](https://pkg.go.dev/search?q=msal)) displays multiple entries to the same project - each with an entirely different API altogether. It would be helpful if the documentation clearly specified the version/branch to use in the `go get` command that is directly associated with the details in the `README.md`.

### PR DESCRIPTION
**Pull request by abhidnya13**
Updated README.md to reflect the recent changes made in the library in #117 
Resolves #142 and #133

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 **MSAL Go is a new addition to the MSAL family of libraries, not yet in public preview, that has been made available to gauge customer interest and to gather early feedback from the community. We welcome all contributors (see [CONTRIBUTING.md](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/CONTRIBUTING.md)) to help us grow our list of supported MSAL SDKs.**
 
-The MSAL library for Go is part of the [Microsoft identity platform for developers](https://aka.ms/aaddevv2) (formerly named Azure AD) v2.0. It enables you to acquire security tokens to call protected APIs. It uses industry standard OAuth2 and OpenID Connect.
```
Comment (henrik-me) [inline README.md:5]: >SAL library for Go is part of the [Microso [](start = 5, length = 42)

This is what we have for all our repo's not sure this should be removed?

File: README.md [hunk 2]
```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
...
-```go
-accounts := publicClientApp.GetAccounts()
-// Assuming the user wanted the first account
-userAccount := accounts[0]
-scopes := []string{"your_scope"}
-silentParams := msal.CreateAcquireTokenSilentParametersWithAccount(scopes, userAccount)
-result, err := publicClientApp.AcquireTokenSilent(silentParams)
-```
+    ```go
+    app, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
```
Comment (henrik-me) [inline README.md:46]: >app [](start = 4, length = 3)

I think this should be named publicClientApp like before, any reason to change?

File: README.md [hunk 3]
```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
...
+    ```
+
+   * Initializing a confidential client:
+
+    ```go
+    // Initializing the client credential
+    cred, err := confidential.NewCredFromSecret("client_secret")
```
Comment (henrik-me) [inline README.md:53]: >NewCredFromSecret [](start = 30, length = 17)

this name is odd, should be as it is in other libraries. New is fine but CredFromSecret sounds wrong. It's the ConfidentialClient, can also remove this and stick to simply New.

File: README.md [hunk 4]
```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
...
+        return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+    }
+    app, err := confidential.New("client_id", cred, confidential.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+1. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
+
+    ```go
+    var userAccount public.Account
+    accounts := app.Accounts()
```
Comment (henrik-me) [inline README.md:64]: what is accounts, and why get a userAccount? is this code even possible? it doesn't look right to me

File: README.md [hunk 5]
```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
...
+    if len(accounts) > 0 {
```
Comment (henrik-me) [inline README.md:65]: is this the code we have in the dev samples as well? perhaps a link to full samples for the same scenarios?

File: README.md [hunk 6]
```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
...
-3. If there is no suitable token in the cache, or you choose to skip this step, now we can send a request to AAD to obtain a token. 
-```go
-if err != nil {
-    // Based on the flow, you create the type of parameters
-    tokenParams := msal.CreateAcquireTokenxxxParameters(scopes, ...)
-    result, err := publicClientApp.AcquireTokenByxxx(tokenParams)
+1. If there is no suitable token in the cache, or you choose to skip this step, now we can send a request to AAD to obtain a token.
+
+    ```go
+    result, err := app.AcquireTokenByOneofTheActualMethods([]string{"your_scope"}, ...(other parameters depending on the function) )
```
Comment (henrik-me) [inline README.md:76]: >ByOneofTheActualMethods [](start = 35, length = 23)

should it be in quotes?

**Review by henrik-me — APPROVED** [review]
:shipit:

File: README.md (hunk 1 above)
Comment (abhidnya13) [reply README.md:5]: Good catch, I was referring to MSAL [Java](https://github.com/AzureAD/microsoft-authentication-library-for-java#microsoft-authentication-library-msal-for-javal) and [Python](https://github.com/AzureAD/microsoft-authentication-library-for-python#microsoft-authentication-library-msal-for-python). So followed a similar format. Just looked at MSAL .net which has it in this format. Adding it back with the link in the next commit

File: README.md (hunk 2 above)
Comment (abhidnya13) [reply README.md:46]: I picked the code snippet from the updated samples. I think `publicClientApp` should be fine too. Although in one of my PRs earlier we shortened something like a `labApplication` to just `app`. Is it a Go convention to have shorter names ? @keegan-caruso , @element-of-surprise ?

File: README.md (hunk 3 above)
Comment (abhidnya13) [reply README.md:53]: `NewCredFromSecret` is a publicAPI [func](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/apps/confidential/confidential.go#L147) to create the credential object. The initialization of the application still uses `confidential.New()`

File: README.md (hunk 5 above)
Comment (abhidnya13) [reply README.md:65]: Yes, have added a link below which points to the dev samples.

File: README.md (hunk 6 above)
Comment (abhidnya13) [reply README.md:76]: Although not entirely runnable code, it still is a sample code snippet so as to distinguish it from the text. I think we should have the quotes.

File: README.md (hunk 3 above)
Comment (henrik-me) [reply README.md:53]: thanks for clarifying, I suppose I object to not sticking more to the MSAL naming to have an easier transition if you know one of our other MSAL's.

File: README.md (hunk 2 above)
Comment (henrik-me) [reply README.md:46]: I still think readable code needs names that indicates what the stuff is for #cleancode :)

---
In reply to: [562163645](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146#discussion_r562163645) [](ancestors = 562163645)

File: README.md (hunk 6 above)
Comment (henrik-me) [reply README.md:76]: I meant: "ByOneOfTheActualMethods" is not quoted afaiks? I think it should be

---
In reply to: [562169548](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146#discussion_r562169548) [](ancestors = 562169548)

File: README.md (hunk 2 above)
Comment (abhidnya13) [reply README.md]: Agreed, updated in the latest [commit](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146/commits/4179b9ba833b42c58ed822ee1f09c2138f1175a6)

File: README.md (hunk 6 above)
Comment (abhidnya13) [reply README.md]: Got it, changed in the latest [commit](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146/commits/4179b9ba833b42c58ed822ee1f09c2138f1175a6)

**Pull request merged**